crates/sqlexec/src/context/local.rs

```diff
@@ -320,6 +324,28 @@ impl LocalSessionContext {
         self.portals.remove(name);
     }
 
+    pub(crate) fn push_notice(&mut self, notice: Notice) {
+        self.notices.push(notice)
+    }
+
+    /// Drain all notices from the session.
+    ///
+    /// This will take into account the 'client_min_messages' session var to
+    /// filter out notices that shouldn't be sent to the client.
+    pub(crate) fn drain_notices(&mut self) -> Vec<Notice> {
```

Comment:
I would have expected this to be `Optional<Vec<Notice>>`. but it sort of seems like it shouldn't even be that, maybe just a bool, or a count of how many there were? if you've drained them,

Reply:
> but it sort of seems like it shouldn't even be that, maybe just a bool, or a count of how many there were? if you've drained them,

I don't understand what this means. `drain_notices` takes the notices from the session so that they can be sent somewhere else higher up (pg protocol or cli). What would bool here do?

Also I picked `Vec<Notice>` since it makes it easier to do `for notice in sess.drain_notices()` elsewhere.

Reply:
I sort of assumed that "drain noticed" did more than just copy them by looking at it so the bool/usize options were really more "just report that you're done" rather than "copy them out".

I think this is fine, and is mostly that "drain" lead me to assume the wrong thing. I'm still a touch worried about constructing an empty vector more often than we need, but otherwise this is fine.

Reply:
I'll rename this to `take_notices` to make it clearer.

Originally had this return an iterator to follow Vec's drain method (https://doc.rust-lang.org/std/vec/struct.Vec.html#method.drain) but changed it since that if makes it a bit harder to return an iterator.